### PR TITLE
change how cosign is built

### PR DESCRIPTION
**Pull request by JasonPowr**
This pr is related to this https://github.com/securesign/image-factory/issues/18, and this pr https://github.com/securesign/cosign/pull/1, and involves changing the build-cosign image to use the correct branch in the securesign/cosign repo

### CODE REVIEW

**Comment by JasonPowr** [comment]
The build-cosign job is currently failing because the release-next branch is not present in the repo, it should be after https://github.com/securesign/cosign/pull/1 is merged and the update-to-head.sh script is run 

This pr should not be merged https://github.com/securesign/cosign/pull/1 is merged

**Comment by lance** [comment]
> The build-cosign job is currently failing because the release-next branch is not present in the repo, it should be after [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged and the update-to-head.sh script is run
> 
> This pr should not be merged [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged

If you rebase this PR, it should pick up the change in https://github.com/securesign/image-factory/pull/20 (which fixes that issue), and the test should pass.

**Comment by JasonPowr** [comment]
> > The build-cosign job is currently failing because the release-next branch is not present in the repo, it should be after [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged and the update-to-head.sh script is run
> > This pr should not be merged [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged
> 
> If you rebase this PR, it should pick up the change in #20 (which fixes that issue), and the test should pass.



> > The build-cosign job is currently failing because the release-next branch is not present in the repo, it should be after [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged and the update-to-head.sh script is run
> > This pr should not be merged [securesign/cosign#1](https://github.com/securesign/cosign/pull/1) is merged
> 
> If you rebase this PR, it should pick up the change in #20 (which fixes that issue), and the test should pass.

All good now Ill go ahead and close that other pr I opened, thanks Lance :)

File: .github/workflows/image-factory.yaml [hunk 1]
```diff
@@ -297,12 +297,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           repository: securesign/cosign
-          ref: ${{ env.COSIGN_VER }}
-
-      - name: Pull down the securesign repo
-        uses: actions/checkout@v2
-        with:
-          path: securesign
+          ref: release-next
```
Comment (lance) [inline .github/workflows/image-factory.yaml]: Once this lands https://github.com/securesign/cosign/pull/1#pullrequestreview-1595758085 and we run the update script as `./redhat/release/update-to-head.sh v2.1.1` to create a `redhat-v2.1.1` branch, this PR should be good to go. But the refs should be to the `redhat-v2.1.1` branch, which can be set in `COSIGN_VER`.

**Comment by JasonPowr** [comment]
Seems to be an issue building the image this way ill investigate

**Review by lance — APPROVED** [review]
/lgtm

**Pull request merged**